src/lib/message/L2ToL1MessageClassic.ts

```diff
@@ -0,0 +1,383 @@
+import { L2ToL1TransactionEvent } from '../abi/ArbSys'
+import { ContractTransaction, Overrides } from 'ethers'
+import { EventFetcher } from '../utils/eventFetcher'
+import {
+  SignerProviderUtils,
+  SignerOrProvider,
+} from '../dataEntities/signerOrProvider'
+import { isDefined, wait } from '../utils/lib'
+import { ArbSdkError } from '../dataEntities/errors'
+import { EventArgs } from '../dataEntities/event'
+import { L2ToL1MessageStatus } from '../dataEntities/message'
+import { getL2Network } from '../dataEntities/networks'
+
+export interface MessageBatchProofInfo {
+  /**
+   * Merkle proof of message inclusion in outbox entry
+   */
+  proof: string[]
+
+  /**
+   * Merkle path to message
+   */
+  path: BigNumber
+
+  /**
+   * Sender of original message (i.e., caller of ArbSys.sendTxToL1)
+   */
+  l2Sender: string
+
+  /**
+   * Destination address for L1 contract call
+   */
+  l1Dest: string
+
+  /**
+   * L2 block number at which sendTxToL1 call was made
+   */
+  l2Block: BigNumber
+
+  /**
+   * L1 block number at which sendTxToL1 call was made
+   */
+  l1Block: BigNumber
+
+  /**
+   * L2 Timestamp at which sendTxToL1 call was made
+   */
+  timestamp: BigNumber
+
+  /**
+   * Value in L1 message in wei
+   */
+  amount: BigNumber
+
+  /**
+   * ABI-encoded L1 message data
+   */
+  calldataForL1: string
+}
+
+/**
+ * Conditional type for Signer or Provider. If T is of type Provider
+ * then L2ToL1MessageReaderOrWriter<T> will be of type L2ToL1MessageReader.
+ * If T is of type Signer then L2ToL1MessageReaderOrWriter<T> will be of
+ * type L2ToL1MessageWriter.
+ */
+export type L2ToL1MessageReaderOrWriterClassic<T extends SignerOrProvider> =
+  T extends Provider ? L2ToL1MessageReaderClassic : L2ToL1MessageWriterClassic
+
+export class L2ToL1MessageClassic {
+  /**
+   * The number of the batch this message is part of
+   */
+  public readonly batchNumber: BigNumber
+
+  /**
+   * The index of this message in the batch
+   */
+  public readonly indexInBatch: BigNumber
+
+  protected constructor(batchNumber: BigNumber, indexInBatch: BigNumber) {
+    this.batchNumber = batchNumber
+    this.indexInBatch = indexInBatch
+  }
+
+  public static fromBatchNumber<T extends SignerOrProvider>(
+    l1SignerOrProvider: T,
+    batchNumber: BigNumber,
+    indexInBatch: BigNumber
+  ): L2ToL1MessageReaderOrWriterClassic<T>
+  public static fromBatchNumber<T extends SignerOrProvider>(
+    l1SignerOrProvider: T,
+    batchNumber: BigNumber,
+    indexInBatch: BigNumber
+  ): L2ToL1MessageReaderClassic | L2ToL1MessageWriterClassic {
+    return SignerProviderUtils.isSigner(l1SignerOrProvider)
+      ? new L2ToL1MessageWriterClassic(
+          l1SignerOrProvider,
+          batchNumber,
+          indexInBatch
+        )
+      : new L2ToL1MessageReaderClassic(
+          l1SignerOrProvider,
+          batchNumber,
+          indexInBatch
+        )
+  }
+
+  public static async getL2ToL1MessageLogs(
```

Comment:
we call the similar function in nitro `getL2ToL1Events`, would be good to use the same func names

Reply:
agreed, updated